sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveInspectors.scala

```diff
@@ -305,12 +305,17 @@ private[hive] trait HiveInspectors {
         withNullSafe(o => getByteWritable(o))
       case _: ByteObjectInspector =>
         withNullSafe(o => o.asInstanceOf[java.lang.Byte])
-      case _: JavaHiveVarcharObjectInspector =>
```

Comment:
do we still need to handle `JavaHiveVarcharObjectInspector`?

Reply:
Sorry for late reply. It is no need. `case _: HiveVarcharObjectInspector =>` has already include this.

I don not known why some type add additional java check like `JavaHiveDecimalObjectInspector`, `JavaDateObjectInspector`. But most type has not, like `String`, `Int`, `Binary`.

So shall we retain `JavaHiveVarcharObjectInspector ` check to respect ?

Reply:
do you mean `JavaHiveVarcharObjectInspector` extends `HiveVarcharObjectInspector`?

Reply:
Yes.